pkgs/tools/networking/bitmask-vpn/default.nix

```diff
@@ -0,0 +1,148 @@
+{ lib
+, fetchFromGitLab
+, mkDerivation
+, buildGoModule
+, wrapQtAppsHook
+, python3Packages
+, pkg-config
+, openvpn
+, cmake
+, qmake
+, which
+, iproute2
+, iptables
+, procps
+, qmltermwidget
+, qtbase
+, qtdeclarative
+, qtquickcontrols
+, qtquickcontrols2
+, qttools
+, provider ? "riseup"
+}:
+let
+  version = "0.21.6";
+
+  src = fetchFromGitLab {
+    domain = "0xacab.org";
+    owner = "leap";
+    repo = "bitmask-vpn";
+    rev = version;
+    sha256 = "sha256-LMz+ZgQVFGujoLA8rlyZ3VnW/NSlPipD5KwCe+cFtnY=";
+  };
+
+  bitmask-root = mkDerivation {
+    inherit src version;
+    sourceRoot = "source/helpers";
+    pname = "bitmask-root";
+    nativeBuildInputs = [ python3Packages.wrapPython ];
+    postPatch = ''
+      substituteInPlace bitmask-root \
+        --replace 'swhich("ip")' '"${iproute2}/bin/ip"' \
+        --replace 'swhich("iptables")' '"${iptables}/bin/iptables"' \
+        --replace 'swhich("ip6tables")' '"${iptables}/bin/ip6tables"' \
+        --replace 'swhich("sysctl")' '"${procps}/bin/sysctl"' \
+        --replace /usr/sbin/openvpn ${openvpn}/bin/openvpn
+      substituteInPlace se.leap.bitmask.policy \
+        --replace /usr/sbin/bitmask-root $out/bin/bitmask-root
+    '';
+    installPhase = ''
+      runHook preInstall
+      install -m 755 -D -t $out/bin bitmask-root
+      install -m 444 -D -t $out/share/polkit-1/actions se.leap.bitmask.policy
+      wrapPythonPrograms
+      runHook postInstall
+    '';
+  };
+in
+
+buildGoModule rec {
+  inherit src version;
+  pname = "${provider}-vpn";
+  vendorSha256 = null;
+
+  postPatch = ''
+    substituteInPlace pkg/helper/linux.go \
+      --replace /usr/sbin/openvpn ${openvpn}/bin/openvpn
+
+    substituteInPlace pkg/vpn/launcher_linux.go \
+      --replace /usr/sbin/openvpn ${openvpn}/bin/openvpn \
+      --replace /usr/sbin/bitmask-root ${bitmask-root}/bin/bitmask-root \
+      --replace /usr/bin/lxpolkit /run/wrappers/bin/polkit-agent-helper-1 \
+      --replace '"polkit-gnome-authentication-agent-1",' '"polkit-gnome-authentication-agent-1","polkitd",'
+
+    # Note that introducing the provider's name into the action's name
+    # is required to let polkit authorize more than one bitmask-root.
+    # On NixOS, actions policies are automatically installed from the package
+    # into /run/current-system/sw/share/polkit-1/actions/
+
+    substituteInPlace pkg/pickle/helpers.go \
+      --replace /usr/share $out/share
+
+    # Using $PROVIDER is not working
+    substituteInPlace providers/vendor.conf \
+      --replace "provider = riseup" "provider = ${provider}"
+
+    patchShebangs gui/build.sh
+    wrapPythonProgramsIn branding/scripts
+  '';
+
+  nativeBuildInputs = [
```

Comment:
it seems that most of those inputs are only used for linux.
Could you mark them as optional for linux for now.
https://0xacab.org/leap/bitmask-vpn/-/blob/main/Makefile#L78

I should have time to test on darwin on monday.

Reply:
@happysalada, I don't think it's reliable to compare Debian's dependencies and Brew's dependencies: a single Brew package may provide the same content provided by multiple Debian packages.
I've read the `Makefile` to see whether the building toolchain would be significantly different on `darwin` and those `darwin`-specific rules appear to be mostly about building an installer (dmg image), but I guess that with Nixpkgs we would not need such installer. But I don't know `darwin` so I may be wrong.
Best is to make it work on `darwin` if you can, and then see if some dependencies are superfluous or broken but optional there.
You may use a developing cycle based upon:

```
rm -rf source
nix -L develop -f . riseup-vpn
unpackPhase
cd source
patchPhase
eval "$buildPhase" || echo ERROR
[hack ... hack ... hack]
```
And that would be `nix -L build -f . riseup-vpn` to get the final `./result/`.
Do not hesitate to post the errors you cannot overcome here, so I can see if I can help.